rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -1,15 +1,136 @@
-pub struct Tun {}
+use connlib_shared::{messages::Interface as InterfaceConfig, Result};
+use ip_network::IpNetwork;
+use std::{
+    io,
+    str::FromStr,
+    sync::Arc,
+    task::{ready, Context, Poll},
+};
+use tokio::sync::mpsc;
+
+const TUNNEL_UUID: &str = "e9245bc1-b8c1-44ca-ab1d-c6aad4f13b9c";
+
+// TODO: Make sure all these get dropped gracefully on disconnect
+pub struct Tun {
+    _adapter: Arc<wintun::Adapter>,
+    // TODO: Get rid of this mutex. It's a hack to deal with `poll_read` taking a `&self` instead of `&mut self`
+    packet_rx: std::sync::Mutex<mpsc::Receiver<wintun::Packet>>,
+    _recv_thread: std::thread::JoinHandle<()>,
+    session: Arc<wintun::Session>,
+}
 
 impl Tun {
-    pub fn new() -> Self {
-        Self {}
+    pub fn new(config: &InterfaceConfig) -> Result<Self> {
+        // The unsafe is here because we're loading a DLL from disk and it has arbitrary C code in it.
+        // As a defense, we could verify the hash before loading it. This would protect against accidental corruption, but not against attacks. (Because of TOCTOU)
+        let wintun = unsafe { wintun::load_from_path("./wintun.dll") }?;
+        let uuid = uuid::Uuid::from_str(TUNNEL_UUID)?;
+        let adapter = match wintun::Adapter::create(
+            &wintun,
+            "Firezone",
+            "Firezone VPN",
```

Comment:
(nit) We try to avoid VPN naming where we can because it we're "Zero Trust"

```suggestion
            "Firezone Tunnel",
```